src/executables/master_heartbeat.c

```diff
@@ -1633,22 +1633,24 @@ hb_cluster_send_heartbeat_internal (struct sockaddr_in *saddr, socklen_t saddr_l
 {
   HBP_HEADER *hbp_header;
   char buffer[HB_BUFFER_SZ], *p;
-  size_t hb_len;
+  const size_t hb_body_len = OR_INT_SIZE * 3;	//heartbeat message body length
+  const size_t hb_len = sizeof (HBP_HEADER) + hb_body_len;	//heartbeat message length including header and body
+
   int send_len;
 
   memset ((void *) buffer, 0, sizeof (buffer));
   hbp_header = (HBP_HEADER *) (&buffer[0]);
 
-  int error_code = hb_set_net_header (hbp_header, HBP_CLUSTER_HEARTBEAT, is_req, OR_INT_SIZE, 0, dest_host_name);
+  int error_code = hb_set_net_header (hbp_header, HBP_CLUSTER_HEARTBEAT, is_req, hb_body_len, 0, dest_host_name);
   if (error_code != NO_ERROR)
     {
       return error_code;
     }
 
   p = (char *) (hbp_header + 1);
   p = or_pack_int (p, hb_Cluster->state);
-
-  hb_len = sizeof (HBP_HEADER) + OR_INT_SIZE;
+  p = or_pack_int (p, hb_Cluster->myself->is_tran_server_alive);
+  p = or_pack_int (p, hb_Cluster->myself->is_page_server_alive);
```

Comment:
Can you reduce the size of the message to transfer to the network? This kind of information of all is enough 32-byte. It would be good to handle later for optimization.

Reply:
It is possible to fit two flags in a four-byte integer, but I will optimize it later.